src/pages/workflows/workflow/common/WorkflowModal.tsx

```diff
@@ -0,0 +1,110 @@
+import { ButtonPrimary, Modal } from '@terra-ui-packages/components';
+import React from 'react';
+import { TextInput } from 'src/components/input';
+import { FormLabel } from 'src/libs/forms';
+
+interface WorkflowModalProps {
+  setCreateWorkflowModalOpen: (x: boolean) => void;
+  title: string;
+  namespace: string;
+  name: string;
+  synopsis: string;
+  buttonAction: string; // name of the primary button i.e. 'save' or 'upload'
+  setWorkflowNamespace: (value: string) => void;
+  setWorkflowName: (value: string) => void;
+  setWorkflowSynopsis: (value: string) => void;
+}
+
+interface NamespaceNameSectionProps {
+  namespace: string | undefined;
+  name: string | undefined;
+  setWorkflowNamespace: (value: string) => void;
+  setWorkflowName: (value: string) => void;
+}
+
+interface SynopsisSnapshotSectionProps {
+  synopsis: string;
+  setWorkflowSynopsis: (value: string) => void;
+}
+
+const NamespaceNameSection = (props: NamespaceNameSectionProps) => {
+  const namingMessage = 'Only letters, numbers, underscores, dashes, and periods allowed';
+  const { namespace, name, setWorkflowNamespace, setWorkflowName } = props;
+
+  return (
+    <>
+      <div style={{ flexWrap: 'wrap', flexGrow: 1, flexBasis: '400px' }}>
+        <div style={{ marginBottom: '0.1667em' }}>
+          <FormLabel>Namespace</FormLabel>
```

Comment:
small nit: what do you think about adding `*` to Namespace and Name (below) to indicate these are required fields similar to how we do in other modals to make it consistent?

![Screenshot 2024-08-22 at 2 27 37 PM](https://github.com/user-attachments/assets/25c824b9-73bf-487c-93a9-af8f268607bb)

Reply:
I can definitely mark these as required